### ISSUE TITLE
Screen Reader announce incorrect information as 'Toggle selection for all items' on toggle selection of a single file or folder.

### ISSUE THREAD

**Issue by otaha2** [microsoft/fluentui#18826]
### Environment Information

- **Tool**: NVDA/JAWS
- **Browser and OS versions**: Edge Version 92.0.902.30 (Official build) beta (64-bit)

### Describe the issue:

Screen reader announces 'Toggle Selection for All Items' upon navigation to or selection of the checkbox of a single item in the list. 

### Please provide a reproduction of the issue in a codepen:

Repro instructions: 
1. Navigate to checkbox of any single item in the table and observe the issue with the announcement 

Reproduced with the basic DetailsList as shown in the below codepen
https://codepen.io/otaha2/pen/PomPqYV

#### Actual behavior:

Screen Reader announces incorrect information as 'Toggle selection for all items' on toggle selection of a single file or folder. 
Screen Reader announces: " Row 8 Toggle selection for all items column 1 checkbox not checked checkbox for test2.xlsx."
Column header name is defined as 'Toggle selection for all items' and due to association screen reader announces same on every file/folder selection.

#### Expected behavior:

Correct and descriptive column header should be defined, so that when it is associated with data cells it won't create confusion for AT users.
Screen Reader should announce: " Row 8 Toggle selection to select text2.xlsx column 1 checkbox not checked checkbox" for test2.xlsx.

### Documentation describing expected behavior

This seams to be caused by the checkbox column header having an aria-labelledby property that points to 'Toggle Selection for All Items' which causes that to be announced upon navigation to the selection for a single item. This can very easily be confusing for a user.

**Comment by gouttierre** [comment]
@otaha2- Thanks for filing this issue with us.

@smhigley -  Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by smhigley** [comment]
@otaha2 the reason the checkbox in the column header has the label "toggle selection for all items" is because that checkbox _does_ toggle selection for all items. If we change the columnheader, that checkbox won't have a descriptive name.

I get the concern, but I'm not sure mis-naming the select-all checkbox is a good solution.
